src/osd/OSD.cc

```diff
@@ -1374,6 +1374,18 @@ void OSDService::got_stop_ack()
   }
 }
 
+void OSDService::build_map(
+int& max,
+ssize_t& max_bytes,
+bufferlist& bl,
+std::function<bool()> map_getter) {
+  if (!map_getter()) {
+    throw -ENOENT;
+  }
+  max--;
+  max_bytes -= bl.length();
+}
```

Comment:
This should be a lambda local to build_incremental_map_message -- every caller passes the same map_getter.

Just return ENOENT, please don't introduce exceptions unnecessarily.

Reply:
This should probably return std::optional<bufferlist>

Reply:
Probably rename to get_map_and_adjust_counters

Reply:
Renamed, made local and found it easier to return a boolean instead.